CRM/Contribute/Form/Contribution/Confirm.php

```diff
@@ -1933,6 +1933,9 @@ public static function submit($params) {
     $_SERVER['REQUEST_METHOD'] = 'GET';
     $form->controller = new CRM_Contribute_Controller_Contribution();
     $params['invoiceID'] = md5(uniqid(rand(), TRUE));
+
+    // We want to move away from passing in amount as it is calculated by the actually-submitted params.
+    $params['amount'] = $params['amount'] ?? $form->getMainContributionAmount($params);
```

Comment:
This submit function is accessed by the ContributionPage.submit api & a whole load of tests via that, but not from the form submission process